gradle_plugin/src/main/groovy/com/microsoft/hydralab/utils/HydraLabClientUtils.java

```diff
@@ -660,7 +654,14 @@ private static JsonObject triggerTestRun(String runningType, HydraLabAPIConfig a
         jsonElement.addProperty("needClearData", apiConfig.needClearData);
         jsonElement.addProperty("testRunnerName", apiConfig.testRunnerName);
         jsonElement.addProperty("testScope", apiConfig.testScope);
-
+        if (apiConfig.neededPermissions != null) {
+            jsonElement.add("neededPermissions", GSON.toJsonTree(apiConfig.neededPermissions));
+        }
+        if (StringUtils.isNotBlank(apiConfig.deviceActionsStr)) {
+            JsonParser parser = new JsonParser();
+            JsonObject jsonObject = parser.parse(apiConfig.deviceActionsStr).getAsJsonObject();
```

Comment:
Please surround with try catch when parse jsonstring. If error happens, log the original String.